src/main/java/com/minecolonies/structures/helpers/StructureProxy.java

```diff
@@ -24,14 +19,12 @@
 public class StructureProxy
 {
     private final Structure structure;
-    private final List<TileEntity> tileEntities = new ArrayList<>();
-    private final List<Entity>     entities     = new ArrayList<>();
-    private Block[][][]       blocks;
-    private IBlockState[][][] metadata;
-    private int               width;
-    private int               height;
-    private int               length;
-    private BlockPos          offset;
+    private Template.EntityInfo[][][] entities;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private field 'entities' could be made final; it is only initialized in the declaration or constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)